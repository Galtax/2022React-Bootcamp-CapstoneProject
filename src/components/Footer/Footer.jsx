import React from 'react';
import PropTypes from 'prop-types';
import {  FooterMain
        , Legend
        , SocialContainer
        , SocialLink
        , IconGithub } from './FooterMain.styled'
import myIconGithub from "../../assets/gitlogo.png"

function Footer({ text }) {
  return (
    <FooterMain>
      <Legend>
        {text}
      </Legend>
      <SocialContainer>
        <SocialLink href="https://github.com/galtax" target={"_blank"}>

          <IconGithub src={myIconGithub} />
        </SocialLink>

      </SocialContainer>
    </FooterMain>
  );
}

Footer.propTypes = {
  text: PropTypes.string.isRequired,
};

export default Footer;
