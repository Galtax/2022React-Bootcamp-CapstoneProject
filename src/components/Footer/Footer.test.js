import React from 'react';
import { mount, shallow } from 'enzyme';
import 'jest-styled-components';
import Footer from '.';

describe('Footer Component Test', () => {

  it('should render the Footer Component correctly', () => {
    const wrapper = mount(<Footer text='Lorem Ipsum' />);

    expect(wrapper).toMatchSnapshot();
  });

  it('should contain subcomponents', () => {
    const mockText = 'Lorem Ipsum';
    let wrapper = shallow(<Footer text='Lorem Ipsum' />);
    
    expect(wrapper.exists()).toEqual(true);
    expect(wrapper.text()).toEqual(mockText);
  });
});
