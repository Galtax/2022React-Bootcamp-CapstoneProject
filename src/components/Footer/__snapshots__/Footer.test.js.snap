// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer Component Test should render the Footer Component correctly 1`] = `ReactWrapper {}`;
